NickvisionTubeConverter.GNOME/Controls/DownloadRow.cs

```diff
@@ -98,6 +103,12 @@ private DownloadRow(Gtk.Builder builder, Guid id, string filename, string saveFo
             _lblLog.GetClipboard().SetText(_lblLog.GetText());
             _sendNotificationCallback(new NotificationSentEventArgs(_localizer["LogCopied"], NotificationSeverity.Informational));
         };
+        _pulsingBarCallback = (data) =>
+        {
+            _pulsingBar.Pulse();
+            return true;
```

Comment:
I don't like that this returns true always, causing this callback to run forever every 30 seconds, never stopping....

Can you call g_timeout_add or g_idle_add everytime SetProgressState is called and return false instead?

Reply:
Added a variable to control whether the callback should be running